### PR TITLE
BookManagement: Update BEAN,DAO,SERVLET for author, category, book.

### PR DESCRIPTION
**Pull request by vuong196**
@hongvo2308 I formatted code following your advices. Could you have a look? Thank you.

### CODE REVIEW

File: src/beans/Author.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	@GeneratedValue(generator = "uuid")
+	@GenericGenerator(name = "uuid", strategy = "uuid2")
+	@Column(name = "author_id", length = 36, unique = true, nullable = false)
+	private String	_authorId;
+	
+	@Column(name = "author_name", length = 50, nullable = false)
+	private String	_authorName;
+	
+	@OneToMany(fetch = FetchType.LAZY, mappedBy = "_bookAuthor")
+	private Set<Book>	_authorBookSet	= new HashSet<>(0);
```
Comment (ces-hongvo) [inline src/beans/Author.java]: Fix the indent

File: src/beans/Author.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+	public Author() {
+
+		this.setAuthorId(UUID.randomUUID().toString());
```
Comment (ces-hongvo) [inline src/beans/Author.java:21]: Please assign value directly

File: src/beans/Author.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	public Author(String authorName) {
+
+		this.setAuthorId(UUID.randomUUID().toString());
+		this.setAuthorName(authorName);
+	}
```
Comment (ces-hongvo) [inline src/beans/Author.java:28]: Assigne value directly

File: src/beans/Book.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name = "author_id", nullable = false)
+	private Author	_bookAuthor;
+	
+	@ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
+	@JoinTable(name = "book_category", catalog = "bookmanagement", joinColumns = {
+		@JoinColumn(name = "book_id", nullable = false, updatable = false) }, inverseJoinColumns = {
+			@JoinColumn(name = "category_id", nullable = false, updatable = false) })
+	private Set<Category>	_bookCategorySet	= new HashSet<>(0);
+
```
Comment (ces-hongvo) [inline src/beans/Book.java]: Fix the indent

File: src/beans/Book.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	public Book(String name, Set<Category> categories, Author author) {
+
+		this.setBookId(UUID.randomUUID().toString());
+		this.setBookAuthor(author);
+		this.setBookCategorySet(categories);
+		this.setBookName(name);
+	}
```
Comment (ces-hongvo) [inline src/beans/Book.java:33]: Assign directly

File: src/beans/Category.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	public Category() {
+
+		this.setCategoryId(UUID.randomUUID().toString());
+	}
+
+	public Category(String categoryName) {
+
+		this.setCategoryId(UUID.randomUUID().toString());
+		this.setCategoryName(categoryName);
+	}
```
Comment (ces-hongvo) [inline src/beans/Category.java:28]: Please assign directly

File: src/daos/AuthorDAO.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+import beans.Author;
+import configurations.HibernateUtils;
+
+public class AuthorDAO {
+
+	public static boolean delete(String id) throws Exception {
+
```
Comment (ces-hongvo) [inline src/daos/AuthorDAO.java:15]: Please explain why we have to use try catch and throws for the same kind of exception

**Review by ces-hongvo — CHANGES_REQUESTED** [review]
Hi @vuong196 
I have left some comments. Please be noted.
Thanks

File: src/beans/Author.java (hunk 3 above)
Comment (vuong196) [reply src/beans/Author.java:28]: I got it. Thank you

File: src/beans/Book.java (hunk 5 above)
Comment (vuong196) [reply src/beans/Book.java:33]: I fixed them, please have a look.

File: src/beans/Category.java (hunk 6 above)
Comment (vuong196) [reply src/beans/Category.java:28]: I fixed them, please have a look.

File: src/daos/AuthorDAO.java (hunk 7 above)
Comment (vuong196) [reply src/daos/AuthorDAO.java:15]: Yeah I deleted unnecessary Exception caught. Please check them for me

File: src/beans/Author.java (hunk 1 above)
Comment (vuong196) [reply src/beans/Author.java]: I fixed it. Please review it

File: src/beans/Book.java (hunk 4 above)
Comment (vuong196) [reply src/beans/Book.java]: I fixed them, please have a look.

**Pull request merged**